ui/app/src/services/configuration.ts

```diff
@@ -156,158 +155,8 @@ export function setActiveTargetingModel(data): Observable<ActiveTargetingModel>
-          if (rightSection) {
-            rightSection.querySelectorAll('widget').forEach((e, index) => e.setAttribute('uiKey', String(index)));
-            config.preview.toolbar.rightSection = applyDeserializedXMLTransforms(deserialize(rightSection), {
-              arrays,
-              renameTable
-            });
-          }
-        }
-        const toolsPanelPages = xml.querySelector('[id="craftercms.components.ToolsPanel"] > configuration > widgets');
-        if (toolsPanelPages) {
-          // When rendering widgets dynamically and changing pages on the tools panel, if there are duplicate react key
-          // props across pages, react may no swap the components correctly, incurring in unexpected behaviours.
-          // We need a unique key for each widget.
-          toolsPanelPages.querySelectorAll('widget').forEach((e, index) => e.setAttribute('uiKey', String(index)));
-          const lookupTables = ['fields'];
-          config.preview.toolsPanel = applyDeserializedXMLTransforms(deserialize(toolsPanelPages), {
-            arrays,
-            lookupTables,
-            renameTable
-          });
-        }
-        const launcher = xml.querySelector('[id="craftercms.components.Launcher"] > configuration');
-        if (launcher) {
-          launcher.querySelectorAll('widget').forEach((e, index) => e.setAttribute('uiKey', String(index)));
-          config.launcher = applyDeserializedXMLTransforms(deserialize(launcher), {
-            arrays,
-            renameTable
-          }).configuration;
-        }
-        const pageBuilderPanel = xml.querySelector(
-          '[id="craftercms.components.PageBuilderPanel"] > configuration > widgets'
-        );
-        if (pageBuilderPanel) {
-          const lookupTables = ['fields'];
-          pageBuilderPanel.querySelectorAll('widget').forEach((e, index) => {
-            e.setAttribute('uiKey', String(index));
-            if (e.getAttribute('id') === 'craftercms.components.ToolsPanelPageButton') {
-              e.querySelector(':scope > configuration')?.setAttribute('target', 'pageBuilderPanel');
-            }
-          });
-          config.preview.pageBuilderPanel = applyDeserializedXMLTransforms(deserialize(pageBuilderPanel), {
-            arrays,
-            renameTable,
-            lookupTables
-          });
-        }
-        const dashboard = xml.querySelector('[id="craftercms.components.Dashboard"] > configuration');
-        if (dashboard) {
-          dashboard.querySelectorAll('widget').forEach((e, index) => e.setAttribute('uiKey', String(index)));
-          config.dashboard = applyDeserializedXMLTransforms(deserialize(dashboard), {
-            arrays,
-            renameTable
-          }).configuration;
-        }
-        return config;
-      } else {
-        return null;
-      }
-    })
-  );
+export function fetchSiteUiConfig(site: string): Observable<string> {
+  return fetchConfigurationXML(site, '/ui.xml', 'studio');
```

Comment:
I thought we were doing the references expansion and uiKey attribute setting here?

Reply:
Per discussion, we'll leave it as it is for now